### ISSUE TITLE
Expected style "marginVertical: 15" to contain units

### ISSUE THREAD

**Issue by sibelius** [stylelint/stylelint#4703]
> What is the problem you're trying to solve?

I'd like to ensure all styles have units in react-native

related to https://github.com/styled-components/css-to-react-native/issues/40

> What solution would you like to see?

A new rule that warnings/errors based on this

this should work well

```jsx
const Wrapper = styled`
   margin-vertical: 15px;
`
```

this should error/warn:
```jsx
const Wrapper = styled`
   margin-vertical: 15;
`
```

**Comment by jeddy3** [comment]
@sibelius Thanks for the request and for using the template.

I think this is an excellent candidate for a [plugin](https://stylelint.io/developer-guide/plugins). You can rustle up something using [react-native-known-styling-properties](https://www.npmjs.com/package/react-native-known-styling-properties) and [postcss-values-parser](https://www.npmjs.com/package/postcss-values-parser), either as a standalone plugin or as part of the [stylelint-react-native plugin pack](https://github.com/kristerkari/stylelint-react-native).

---

If anyone runs into this issue in the future wondering how to disallow unitless values for standard CSS properties, you can do that using the [stylelint-csstree-validator plugin](https://www.npmjs.com/package/stylelint-csstree-validator) which will validate your values against the CSS specifications.
